src/main/java/io/debezium/connector/vitess/VitessValueConverter.java

```diff
@@ -274,9 +276,15 @@ private Object convertEnumToString(List<String> options, Column column, Field fi
      */
     protected Object convertSetToString(List<String> options, Column column, Field fieldDefn, Object data) {
         return convertValue(column, fieldDefn, data, "", (r) -> {
-            // The binlog will contain a long with the indexes of the options in the set value ...
-            long indexes = ((Long) data).longValue();
-            r.deliver(convertSetValue(column, indexes, options));
+            if (data instanceof String) {
+                // The value has already been converted to a string so deliver as is
+                r.deliver(data);
+            }
+            else {
+                // The binlog will contain a long with the indexes of the options in the set value ...
```

Comment:
We should actually be okay. We still read the sign bit as part of the bitmask normally later on. As long as we have a 64 bit type to fit all the bits in (even if one is a sign bit), it will work properly.

However, some things could be clarified.

Updated code comment.

Also in convertSetValue we use mod 2 operator. This assumes all even unsigned are even signed, and same for odd. Although this assumption is true, we don't need it and can use bitwise operators only to not rely on the numerical interpretation.

Reply:
OK, cool. Just wanted to be sure. As long as we look at / consider all 64 bits/digits later on (I didn't see that part) then we're good.